frontend/src/component/ui/SearchBar.spec.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { SearchBar } from './SearchBar';
+
+function getSearchBarUtil() {
+  const changeHandler = jest.fn();
+  const searchBarRef = React.createRef<SearchBar>();
+  const { getByPlaceholderText } = render(
+    <SearchBar
+      ref={searchBarRef}
+      onChange={changeHandler}
+      autoCompleteSuggestions={[
+        {
+          originalUrl: 'https://www.google.com/',
+          alias: 'google'
+        },
+        {
+          originalUrl: 'https://github.com/short-d/short/',
+          alias: 'short'
+        }
```

Comment:
Each test should have its own data.

Reply:
Instead of creating a helper, You can leave this code inside `describe` > `beforeEach`.

Reply:
@byliuyang, [these lines](https://github.com/short-d/short/blob/5e1e3443a200eff204a82a0114fbca7b3b1db617/frontend/src/component/ui/SearchBar.tsx#L47-L49) would not be tested if i use beforeEach...any way to make exception for it (i can create 2 test suites to solve this problem)